kornia/feature/orientation.py

```diff
@@ -0,0 +1,149 @@
+from typing import Tuple
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import math
+from kornia.filters import get_gaussian_kernel2d
+from kornia.filters import SpatialGradient
+from kornia.geometry import pi, angle_to_rotation_matrix
+from kornia.feature import (extract_patches_from_pyramid, make_upright,
+                            normalize_laf, raise_error_if_laf_is_not_valid)
+from kornia.geometry import rad2deg
+
+
+class PassLAF(nn.Module):
+    """Dummy module to use instead of local feature orientation or affine shape estimator
+    """
+    def forward(self, laf: torch.Tensor, img: torch.Tensor) -> torch.Tensor:  # type: ignore
+        """
+        Args:
+            laf: torch.Tensor: 4d tensor
+            img (torch.Tensor): the input image tensor
+
+        Return:
+            torch.Tensor: unchanged laf from the input."""
+        return laf
+
+
+class PatchDominantGradientOrientation(nn.Module):
+    """Module, which estimates the dominant gradient orientation of the given patches, in radians.
+    Zero angle points towards right.
+    Args:
+            patch_size: int, default = 32
+            num_angular_bins: int, default is 36
+            eps: float, for safe division, and arctan, default is 1e-8"""
+    def __init__(self,
+                 patch_size: int = 32,
+                 num_angular_bins: int = 36, eps: float = 1e-8):
+        super(PatchDominantGradientOrientation, self).__init__()
+        self.patch_size = patch_size
+        self.num_ang_bins = num_angular_bins
+        self.gradient = SpatialGradient('sobel', 1)
+        self.eps = eps
+        self.angular_smooth = nn.Conv1d(1, 1, kernel_size=3, padding=2, bias=False, padding_mode="circular")
+        with torch.no_grad():
+            self.angular_smooth.weight[:] = torch.tensor([[[0.33, 0.34, 0.33]]])
```

Comment:
wouldn't be more clean with F.conv1d and just setting the kernel here ?

Reply:
True.